lib/perl/Genome/Model/Tools/Sx/Trim/ByPosition.pm

```diff
@@ -0,0 +1,140 @@
+package Genome::Model::Tools::Sx::Trim::ByPosition;
+
+use strict;
+use warnings;
+
+use Genome;
+
+use Genome::Model::Tools::Sx::Split::ByNs;
+use Params::Validate (qw/ :types validate_pos /);
+use Regexp::Common;
+use Try::Tiny;
+
+class Genome::Model::Tools::Sx::Trim::ByPosition {
+    is => 'Genome::Model::Tools::Sx::Base',
+    has => {
+        positions_path => {
+            is => 'File',
+            doc => 'Path to he file of positions to trim.',
+        },    
+    },
+    has_transient_optional => {
+        trim_positions => { is => 'HASH', },
+    },
+};
+
+sub help_brief { "Trim sequences by positions" }
+sub help_detail { help_brief() }
```

Comment:
Ah - no. This is a lift over a Kyung script. I deplore the format, but it's consistency was insisted upon. I will add docs.